vapi/esx/settings/clusters/vms/transition.go

```diff
@@ -0,0 +1,220 @@
+	SourceVmSelectionSpecs map[string]*VmSelectionSpec `json:"source_vm_selection_specs,omitempty"`
+
+	// Cluster module to be reused for transitioned System VMs. Used to express
+	// VM-VM anti affinity relation between System VMs in the vSphere Cluster.
+	// The module must exist for the cluster where the solution is installed.
+	//
+	// If unset, no cluster module is reused. vLCM creates a new module if needed.
+	ClusterModule string `json:"cluster_module,omitempty"`
+}
+
+// Enable enables an EAM managed solution in vLCM. The solution specification is
+// validated before the enablement is started.
+//
+// The enablement only transfers ownership of the solution from EAM to LCCM
+// and sets the desired state in LCCM. The new desired state is not applied,
+// the solution system VMs are untouched.
+//
+// The following happens once the operation is started:
+// - A removal of the corresponding agency in EAM is triggered.
+//
+// The following happens once the operation is completed:
+// - The corresponding agency in EAM can no longer be controlled through the EAM API.
+// - The management of the desired solution specification can be done only through vLCM. See Solutions
+//
+// @param cluster Identifier of the cluster.
+// @param solution Identifier of the solution.
+// @param spec Enablement specification.
+//
+// @throws Error If there is an internal error. The accompanying error message will give more details about the failure.
+// @throws InvalidArgument If the validation of the cluster desired specification fails, the solution associated with solution already exists in vLCM for the cluster, or the agency associated with the eamAgencyID exists, but is not in the scope of the cluster. The value of the data field of Error contains more information. It is a structure that contains all the fields defined in ValidationResult.
+// @throws NotAllowedInCurrentState If System VMs are disabled on the cluster via internal ESX Agent Manager (EAM) API (eam.EsxtAgentManager#disable).
+// @throws Unsupported If the cluster associated with cluster is not managed by vLCM.
+// @throws NotFound If there is no cluster associated with the cluster or if there is no EAM agency associated with the eamAgencyID field of the spec in the system.
+// @throws ServiceUnavailable If the service is not available.
+// @throws Unauthenticated If the caller is not authenticated.
+// @throws Unauthorized If the user doesn't have the required privileges.
+func (m *Manager) Enable(ctx context.Context, cluster types.ManagedObjectReference, solution string, spec *EnableSpec) error {
+	p := clusterTransitionPath(cluster).String()
+	url := m.Resource(p).WithSubpath(solution).WithParam("action", "enable").WithParam("vmw-task", "true")
+	var taskId string
+
+	if err := m.Do(ctx, url.Request(http.MethodPost, spec), &taskId); err != nil {
+		return err
+	}
+
+	_, err := tasks.NewManager(m.Client).WaitForCompletion(ctx, taskId)
```

Comment:
Will this actually return the error if the task has an error?  If I read this right, we get a task but we don't know if the task status is an error.  It's lost forever.

This is why I added a method WaitForRunningOrError.  it waits on the task status to change to RUNNING or  ERROR before returning the error message as an `error`.

Reply:
sure, changed to that